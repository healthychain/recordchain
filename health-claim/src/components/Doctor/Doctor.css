.record-select {
  align-content: center;
  margin: 0 auto;
  margin-top: 3em;
  margin-left:0 auto;
  text-align: center
}

.search-button {
  margin-top: 3em;
  width: 13em;
  height: 2.2em;
  color: black;
}

.bg-search-button {
  background-color:white;
  width: 25em;
  align-self: auto;
  border-color: #0076ba;;
  border-width: 0.07em;
  border-style: solid;
}

.bg-search-button:hover {
  background-color:rgb(140, 180, 220);
  margin: 2em black;   
}

.patient-display {
  margin-top: 2em;
}
