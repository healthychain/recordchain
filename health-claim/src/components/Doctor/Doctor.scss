.dashboard-inner {
  align-content: center;
  margin: 0 auto;
  margin-top: 3em;
  margin-left: 0 auto;
  text-align: center;
  &-alt {
    padding-top: 3em;
    min-width: 800px;
  }
}
.dashboard-main {
  margin: 0 auto;
  max-width: 960px;
}

.Form__Rack {
  padding-bottom: 16px;
  // border-bottom: 1px solid #f0f0f0;
  margin-bottom: 16px;
}

.Box__Tabs {
  display: flex;
  flex-direction: row;
  border-bottom: 1px solid #f0f0f0;
}

.Tab__single {
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 25px;
  padding: 0 3px;
}

.Tab__single__active {
  border-bottom: 3px solid rgba(48, 94, 212, 1);
}

.Tab__text {
  padding-bottom: 2px;
  margin-bottom: 0;
  font-weight: 500;
  color: grey;
}

.Tab__text__active {
  color: rgba(48, 94, 212, 1);
  font-weight: 700;
}

.doctor__content {
  margin-top: 25px;
}

.doctor-layout {
  display: flex;
  height: calc(100vh - 56px - 56px);
  flex-direction: row;
  justify-content: space-between;
}

.search-button {
  margin-top: 3em;
  width: 13em;
  height: 2.2em;
  color: black;
}

.bg-search-button {
  background-color: white;
  width: 25em;
  align-self: auto;
  border-color: #0076ba;
  border-width: 0.07em;
  border-style: solid;
}

.bg-search-button:hover {
  background-color: rgb(140, 180, 220);
  margin: 2em black;
}

.patient-display {
  margin-top: 2em;
}

.separator {
  height: 1px;
  width: 100%;
  border-bottom: 1px solid #f0f0f0;
  margin: 20px 0;
}
