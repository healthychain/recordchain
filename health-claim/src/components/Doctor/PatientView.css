.patient-view-container {
  display: flex;
  flex-direction: row;
}

.patient-img {
  width: 150px;
}
.patient-view-column {
  padding: 20px;
  justify-content: center;
  flex-direction: column;
  align-content: center;
  width: 50%;
}
