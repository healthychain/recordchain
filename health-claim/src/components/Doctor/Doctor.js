import React, { Component } from "react";
import {
  ButtonDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from "reactstrap";
import "./Doctor.css";
import Header from "../Header/Header";
import PatientViewContainer from "../../containers/PatientViewContainer";

class Doctor extends Component {
  constructor(props) {
    super(props);

    this.state = {
      dropdownOpen: false
    };

    this.patientSuggestion = this.patientSuggestion.bind(this);
    this.toggleDropdown = this.toggleDropdown.bind(this);
  }

  patientSuggestion = patient => {
    const { name, id } = patient;
    return (
      <div key={id}>
        <DropdownItem
          onClick={() => {
            this.props.selectPatient({ id, name });
          }}
        >
          {name + " - ID" + id}
        </DropdownItem>
      </div>
    );
  };

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.props.id !== nextProps.id ||
      this.state.dropdownOpen !== nextState.dropdownOpen
    );
  }

  toggleDropdown() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  render() {
    const mockPatients = [
      { name: "John Cena", id: 1 },
      { name: "Andrej Kiska", id: 2 },
      { name: "Jeorrej Olasxzu", id: 3 }
    ];

    return (
      <div>
        <Header />
        <div className="record-select">
          <ButtonDropdown
            className="search-main"
            isOpen={this.state.dropdownOpen}
            toggle={this.toggleDropdown}
            onClick={() => this.toggleDropdown()}
          >
            <DropdownToggle block color="primary" className="search-main" caret>
              {this.props.name ? this.props.name : "Select a patient"}
            </DropdownToggle>
            <DropdownMenu className="search-main">
              {mockPatients.map(this.patientSuggestion)}
            </DropdownMenu>
          </ButtonDropdown>
        </div>
        <div className="patient-display">
          {this.props.id ? (
            <PatientViewContainer
              name={this.props.name}
              id={this.props.id}
              birthDate={this.props.birthDate}
            />
          ) : (
            <div>No patient selected</div>
          )}
        </div>
      </div>
    );
  }
}

export default Doctor;
