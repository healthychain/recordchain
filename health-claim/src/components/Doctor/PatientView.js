import React, { Component } from "react";
import PropTypes from "prop-types";
import "./PatientView.css";

class PatientView extends Component {
  componentDidMount() {
    const { id } = this.props;
    this.props.fetchClaims(id);
  }

  componentWillReceiveProps(newProps) {
    const { id } = this.props;
    const newId = newProps.id;

    if (newId !== id) {
      this.props.fetchClaims(newId);
    }
  }

  render = () => {
    return (
      <div className="patient-view-container">
        {this.props.loading ? (
          <p>Loading</p>
        ) : (
          <div>
            <div className="patient-view-column">
              {" "}
              <img
                alt="patient"
                className="patient-img"
                src="http://www.rw-designer.com/icon-image/14771-256x256x32.png"
              />
              <h4>Name: {this.props.name}</h4>
              <h4>ID: {this.props.id}</h4>
              <h4>
                Birthdate:
                {" 23. 4. 1977"}
              </h4>
            </div>
            <div className="patient-view-column">
              {this.props.claims.map(claim => (
                <p>{claim.title}</p>
              ))}
            </div>
          </div>
        )}
      </div>
    );
  };
}

PatientView.propTypes = {
  loading: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  birthDate: PropTypes.instanceOf(Date).isRequired,
  claims: PropTypes.array
};

PatientView.defaultProps = {
  claims: []
};

export default PatientView;
