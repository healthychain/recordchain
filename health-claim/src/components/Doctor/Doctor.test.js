import React from "react";
import renderer from "react-test-renderer";
import Doctor from "./Doctor";

describe("Doctor", () => {
  it("should render correctly with mocked records", () => {
    const tree = renderer.create(<Doctor loggedIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
