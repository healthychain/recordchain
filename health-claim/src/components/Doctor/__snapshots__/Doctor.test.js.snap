// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor should render correctly with mocked records 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar navbar-expand-md navbar-light"
    >
      <ul
        className="nav"
      >
        <li
          className="nav-item"
        >
           
          <img
            alt="logo"
            src="./icon.png"
          />
        </li>
        <li
          className="header nav-item"
        >
          HEALTH CLAIMS
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="record-select"
  >
    <div
      className="search-main btn-group"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        aria-label={null}
        className="search-main dropdown-toggle btn btn-primary btn-block"
        onClick={[Function]}
        type="button"
      >
        Select a patient
      </button>
      <div
        aria-hidden={true}
        className="search-main dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <div>
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            John Cena - ID1
          </button>
        </div>
        <div>
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            Andrej Kiska - ID2
          </button>
        </div>
        <div>
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            Jeorrej Olasxzu - ID3
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="patient-display"
  >
    <div>
      No patient selected
    </div>
  </div>
</div>
`;
